libexec/scoop-download.ps1

```diff
@@ -0,0 +1,127 @@
+# Usage: scoop download <app> [options]
+# Summary: Download apps in the cache folder
+# Help: e.g. The usual way to download an app, without installing it (uses your local 'buckets'):
+#      scoop download git
+#
+# To download an app from a manifest at a URL:
+#      scoop download https://raw.githubusercontent.com/ScoopInstaller/Main/master/bucket/runat.json
+#
+# To download an app from a manifest on your computer
+#      scoop download path\to\app.json
+#
+# Options:
+#   -f, --force               Force download (overwrite cache)
+#   -s, --skip                Skip hash validation (use with caution!)
+#   -a, --arch <32bit|64bit>  Use the specified architecture, if the app supports it
+
+. "$psscriptroot\..\lib\manifest.ps1"
+. "$psscriptroot\..\lib\install.ps1"
+. "$psscriptroot\..\lib\help.ps1"
+. "$psscriptroot\..\lib\getopt.ps1"
+
+reset_aliases
+
+$opt, $apps, $err = getopt $args 'fsa:' 'force', 'skip', 'arch='
+if ($err) { error "scoop download: $err"; exit 1 }
+
+$check_hash = !($opt.s -or $opt.skip)
+$use_cache = !($opt.f -or $opt.force)
+$architecture = default_architecture
+try {
+    $architecture = ensure_architecture ($opt.a + $opt.arch)
+} catch {
+    abort "ERROR: $_"
+}
+
+if (!$apps) { error '<app> missing'; my_usage; exit 1 }
+
+if (is_scoop_outdated) {
+    scoop update
+}
```

Comment:
Maybe not force scoop update is better?

Reply:
Should I add an additional flag `-u, --update => update scoop before downloading`?

Reply:
Maybe `--update-scoop`?

Reply:
I have gone with the default behaviour of `scoop install` (enforce update). The flag allows to skip it.